src/main/java/cuchaz/enigma/Enigma.java

```diff
@@ -56,7 +55,18 @@ public EnigmaProject openJar(Path path, ProgressListener progress) throws IOExce
 
 		services.get(JarIndexerService.TYPE).forEach(indexer -> indexer.acceptJar(classCache, jarIndex));
 
-		return new EnigmaProject(this, classCache, jarIndex);
+		MessageDigest digest;
+		try {
+			digest = MessageDigest.getInstance("MD5");
+		} catch (NoSuchAlgorithmException e) {
+			throw new RuntimeException(e);
+		}
+		try (InputStream in = new DigestInputStream(Files.newInputStream(path), digest)) {
+			byte[] buffer = new byte[8192];
+			while (in.read(buffer) != -1)
+				;
```

Comment:
Maybe `{}` to make the code more... beautiful?

Reply:
this can be done on one line right?

Reply:
Kinda weird to use an InputStream like this - creating an InputStream that has side effects when you read from it, and then ignoring all the data just to get the side effects. You could use the FileInputStream directly and `digest.update`. The loop would be a few lines longer.